packages/query/src/graphql/plugins/GetMetadataPlugin.ts

```diff
@@ -49,28 +49,31 @@ async function fetchFromApi(): Promise<void> {
 
 async function fetchFromTable(context: any, schemaName: string): Promise<Metadata> {
   const metadata: Metadata = {
-    lastProcessedHeight: undefined,
-    lastProcessedTimestamp: undefined,
-    targetHeight: undefined,
-    chain: undefined,
-    specName: undefined,
-    genesisHash: undefined,
-    indexerHealthy: undefined,
-    indexerNodeVersion: undefined,
-    queryNodeVersion: undefined,
+    lastProcessedHeight: 0,
```

Comment:
Why do you make this change? The gql schema for Metadata allows these fields to be undefined.

Reply:
this way I can type check what value I expect it to be when I pull it from the database.

Reply:
Not a good way. You can create a map to link a metadata field with the expected type in that case.